Sources/XcodeGen/Logger.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import Rainbow
+
+/// Mechanism for logging informational, success, and fatal messages.
+struct Logger {
+
+    // MARK: - Properties
+
+    /// Should informational and success messages be suppressed
+    let isQuiet: Bool
+
+    /// Should the output be colored
+    let isColored: Bool
+
+    // MARK: - Initializers
+
+    /// Initialize a logger
+    ///
+    /// - parameter isQuiet: should informational and success messages be suppressed
+    /// - parameter isColored: should the output be colored
+    init(isQuiet: Bool = false, isColored: Bool = true) {
+        self.isQuiet = isQuiet
+        self.isColored = isColored
+    }
+
+    // MARK: - Logging
+
+    /// Log a fatal message and exit with status code 1
+    ///
+    /// - parameter message: the message to log
+    func fatal(_ message: String) -> Never {
+        print(isColored ? message.red : message)
+        exit(1)
```

Comment:
Seems a bit weird to put this kind of logic in a `Logger`? You wouldn't really expect the program to `exit` after logging some `fatal` error, seems like that should happen in the catch block?

Reply:
Fair. I was just excited to use `-> Never`. I'll remove. I agree it's weird for the logger to `exit`.